sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs

```diff
@@ -0,0 +1,182 @@
+            _mockTokenCredential.Verify(tc => tc.GetTokenAsync(It.IsAny<TokenRequestContext>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public async Task EntraTokenCredential_GetToken_MultipleCallsReturnsCachedToken()
+        {
+            // Arrange
+            var options = CreateEntraTokenCredentialOptions();
+            var mockTransport = CreateMockTransport(200, TokenResponse);
+            var entraTokenCredential = new EntraTokenCredential(options, mockTransport);
+
+            // Act
+            var token = await entraTokenCredential.GetTokenAsync(CancellationToken.None);
+            for (var i = 0; i < 10; i++)
+            {
+                token = await entraTokenCredential.GetTokenAsync(CancellationToken.None);
+            }
+            // Assert
+            Assert.AreEqual(SampleToken, token.Token);
+            _mockTokenCredential.Verify(tc => tc.GetTokenAsync(It.IsAny<TokenRequestContext>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public void EntraTokenCredential_GetToken_ThrowsFailedResponse()
+        {
+            // Arrange
+            var options = CreateEntraTokenCredentialOptions();
+            var errorMessage = "{\"error\":{\"code\":\"BadRequest\",\"message\":\"Invalid request.\"}}";
+            var mockTransport = CreateMockTransport(400, errorMessage);
+            var entraTokenCredential = new EntraTokenCredential(options, mockTransport);
+
+            // Act & Assert
+            Assert.ThrowsAsync<RequestFailedException>(async () => await entraTokenCredential.GetTokenAsync(CancellationToken.None));
+        }
+
+        [Test]
+        public void EntraTokenCredential_GetToken_ThrowsInvalidJson()
+        {
+            // Arrange
+            var options = CreateEntraTokenCredentialOptions();
+            var errorMessage = "{\"error\":{\"code\":\"BadRequest\",\"message\":\"Invalid request.\"}}";
+            var mockTransport = CreateMockTransport(200, errorMessage);
+
+            var entraTokenCredential = new EntraTokenCredential(options, mockTransport);
+
+            // Act & Assert
+            Assert.ThrowsAsync<RequestFailedException>(async () => await entraTokenCredential.GetTokenAsync(CancellationToken.None));
+        }
+
+        [Test]
+        public void EntraTokenCredential_GetToken_RetriesThreeTimesOnTransientError()
+        {
+            // Arrange
+            var options = CreateEntraTokenCredentialOptions();
+            var lastRetryErrorMessage = "Last Retry Error Message";
+            var mockResponses = new MockResponse[]
+            {
+                new MockResponse(500, "First Error Message"),
+                new MockResponse(500, "Second Error Message"),
+                new MockResponse(500, "Third Error Message"),
+                new MockResponse(500, lastRetryErrorMessage),
+                new MockResponse(500, "Fifth Error Message"),
```

Comment:
Nice idea! Maybe let's change the fifth to "Fifth Error Message, shouldn't reach here" to make it very clear?